docs/user-guide/rug/conventions.md

```diff
@@ -101,288 +70,262 @@ the [`CHANGELOG.md` format][changelog].
-similarly to Rug parameters.
+### Parameterizing
 
-### Rug DSL: Avoiding `Begin` and `End` when there is only a Single `do` Action
+Rug parameters are part of the public contract for invoking that Rug. They should follow [snake_case][snake] rules and describe, as explicitly as possible, what the parameter is to contain.
 
-If your `with` statement only has a single corresponding `do` action
-then you should omit the `begin` and `end` block demarcations.
+[snake]: https://en.wikipedia.org/wiki/Snake_case
 
-For example the following is legal but the `begin` and `end`
-statements are not required:
+#### Declaration
 
-```rug
-with file when path = "README.md"
-  begin
-    do replace "{{creation_date}}" { new Date().toISOString().split('T')[0] }
-  end
-```
+Parameters are defined as instance-level variables of the class implementing your Rug. They are decorated with the `#!typescript @Parameter` [decorator][decorators] to declare their metadata. 
 
-It is preferable to omit the `begin` and `end` statements entirely
-when there is only a single associated `do` action:
+That parameter takes a list of properties among the followings:
 
+```typescript
+displayName: string
+description: string
+pattern: string
+validInput: string
+minLength: number = -1
+maxLength: number = -1
+required: boolean = false
 ```
-with file when path = "README.md"
-  do replace "{{creation_date}}" { new Date().toISOString().split('T')[0] }
+
+!!! note
+    Of the given properties, only the `#!typescript pattern` property is mandatory. 
+
+The `#!typescript @Parameter` decorator must be imported into your Rug like this:
+
+```typescript
+import { Parameter } from "@atomist/rug/operations/Decorators";
 ```
 
-### Rug DSL: Indenting Blocks with `Begin` and `End`
-
-When multiple actions are being applied it is important to nest those
-`do` actions within a `begin` and `end` block.  It is conventional to
-place the `begin` indented two spaces on the next line and then to
-indent the `do` statements two more spaces before a closing `end`
-statement indented at the same level as the `begin`:
-
-```rug
-with pom p when path = "pom.xml"
-  begin
-    do setArtifactId  artifact_id
-    do setGroupId group_id
-    do setVersion version
-    do setProjectName name
-    do setDescription description
-  end
+A canonical parameter looks like this:
+
+```typescript
+@Parameter({
+    displayName: "",
+    description: "",
+    pattern: "",
+    validInput: "",
+    minLength: 1,
+    maxLength: 100,
+    required: true|false
+})
+readme_path: string = "README.md";
 ```
 
-### Rug DSL: Labelling Selections Only When They Are Used
+#### Descriptions and Display Names
 
-Sometimes when selecting a particular Rug Type, such as `pom` or
-`file`, it is important to expose what has been selected using a
-label, such as `p`. For example:
+As Rug parameters are part of the public contract to the editor,
+generator, reviewers, executor or predicate it is recommended that a
```

Comment:
Also, we want to provide human-readable name and description so the UI can provide the information to the user.